tests/core/helpers.py

```diff
@@ -12,7 +18,8 @@ def new_transaction(
         private_key,
         gas_price=10,
         gas=100000,
-        data=b''):
+        data=b'',
+        signed=True):
```

Comment:
Instead of `signed=True` we could instead make `private_key` optional and only sign it if the `private_key is not None`.